gax-java/gax/src/main/java/com/google/api/gax/rpc/EndpointContext.java

```diff
@@ -285,9 +303,43 @@ private String determineEndpoint() throws IOException {
             "mTLS is not supported in any universe other than googleapis.com");
       }
 
+      if (shouldUseS2A()) {
+        return mtlsEndpoint();
+      }
```

Comment:
If S2A is supposed to be used, we don't need to go through resolving logic, right? Perhaps we can move this up to the top?

Reply:
Actually, on second though... I think let's keep it here. Universe domain support with mtls may require this additional logic. Let's keep it here.

Reply:
I think this is fine to do while the feature is experimental. Long term, I think we do need to be plumbing the mtls endpoint to the channel creation logic. I pinged with more details. Done in https://github.com/googleapis/sdk-platform-java/pull/3548/commits/9fb0b49fd6c2afffc9b19f9f5b9b852730c26a7e